tensorflow/python/kernel_tests/batch_gemm_op_test.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for tensorflow.ops.tf.BatchGemm."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+
+from tensorflow.python import tf2
+from tensorflow.python.client import session
+from tensorflow.python.compat import compat
+from tensorflow.python.framework import ops
+from tensorflow.python.ops import array_ops
+from tensorflow.python.ops import gradient_checker_v2
+from tensorflow.python.ops import math_ops
+from tensorflow.python.ops import variables
+from tensorflow.python.platform import benchmark
+from tensorflow.python.platform import test
+
+def GetRandomNormalInput(shape, dtype):
+  # float16 has limited range so we reduce the variance of the scalars.
+  scale = 10.0 if dtype != np.float16 else 0.1
+  loc = -10.0 if dtype != np.float16 else 0.1
+  vals = np.array(np.random.normal(loc, scale, np.prod(shape)), dtype=dtype)
+  if dtype in (np.complex64, np.complex128):
+    imag = np.array(np.random.normal(loc, scale, np.prod(shape)), dtype=dtype)
+    vals += 1j * imag
+  return vals.reshape(shape)
+
+
+class BatchGemmOpTest(test.TestCase):
+
+  # Uses numpy to compute batch_matmul(x, y, transpose_a, transpose_b).
+  def _npBatchGemm(self, x, y, transpose_a, transpose_b):
+    # output's shape depends on adj[0] and adj[1]
+    if transpose_a:
+      x = np.conjugate(np.swapaxes(x, -1, -2))
+    if transpose_b:
+      y = np.conjugate(np.swapaxes(y, -1, -2))
+    return np.matmul(x, y)
+
+  # Compares TensorFlow BatchGemm with NumPy's matmul.
+  def _compare(self, x_in, y_in, transpose_a, transpose_b, static_shape):
+    x_t_shape = x_in.shape[:-2] + (x_in.shape[-1], x_in.shape[-2])
+    y_t_shape = y_in.shape[:-2] + (y_in.shape[-1], y_in.shape[-2])
+    x = x_in if not transpose_a else x_in.reshape(x_t_shape)
+    y = y_in if not transpose_b else y_in.reshape(y_t_shape)
+    is_floating = x.dtype != np.int32
+    tol = 100 * np.finfo(x.dtype).eps if is_floating else 0
+    with self.cached_session(use_gpu=is_floating) as sess:
+      # Note: Testing with three dimensions only now
+      z_in = np.ones((x_in.shape[0], x_in.shape[1], y_in.shape[2])).astype(x.dtype)
+      alpha = 1.0
```

Comment:
@jerryyin would the test case be too loose? it seems we are still fixing `alpha` be `1.0` and `beta` be `0.0`?

Reply:
In forward pass, I update both alpha and beta be random floats. In the backward pass, I make alpha random float and beta zero.